### PR TITLE
Refactory kafka pubsub components for code reuse

### PR DESCRIPTION
**Pull request by skyao**
# Description

Refactory kafka pubsub components, extract the kafka common code from pubsub component and laster we can reuse them in kafka binding component.

This is just move code from kafka pubsub to internal/kafka and split the common kafka logic and pubsub component api, it should be safe from the view of functionality.

## Issue reference

https://github.com/dapr/components-contrib/issues/1305

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by shubham1172 — APPROVED** [review]
LGTM overall - had a few questions.

File: pubsub/kafka/kafka.go [hunk 1]
```diff
@@ -15,670 +15,78 @@ package kafka
...
-			consumer.k.logger.Infof("Successfully processed Kafka message after it previously failed: %s/%d/%d [key=%s]", message.Topic, message.Partition, message.Offset, asBase64String(message.Key))
-		}); err != nil {
-			return err
-		}
-	}
-
-	return nil
+type PubSub struct {
+	kafka  *kafka.Kafka
+	topics map[string]bool
```
Comment (shubham1172) [inline pubsub/kafka/kafka.go:27]: We can move `topics` to the Kafka internal implementation too, right?

Comment (shubham1172) [inline pubsub/kafka/kafka.go:27]: Can we move `topics` to the Kafka internal implementation too?

Comment (shubham1172) [inline pubsub/kafka/kafka.go:27]: Do we need a map? Will `[]string` not work?

File: internal/component/kafka/auth.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	"github.com/Shopify/sarama"
+)
+
+func updatePasswordAuthInfo(config *sarama.Config, saslUsername, saslPassword string) {
+	config.Net.SASL.Enable = true
```
Comment (daixiang0) [inline internal/component/kafka/auth.go:26]: Do we support set SASL to disable?

**Comment by skyao** [comment]
This PR is included in PR 

https://github.com/dapr/components-contrib/pull/1696

So close this one.

**Pull request closed without merging**